cmake/CUDAConfig.cmake

```diff
@@ -0,0 +1,32 @@
+if(NOT ${CUDA_ARCH} STREQUAL "")
+	#SET(CUDA_SEPARABLE_COMPILATION ON) # for -dc; no need
+	find_package(CUDA)
```

Comment:
I think you are getting a bit confused between the FindCUDA module (which was how CMake supported CUDA in versions earlier than 3.10), and the CMake CUDA language (first-class CUDA support for CMake that was added in CMake 3.10).  
`find_package(CUDA)` is part of the former method.  However, to use FindCUDA, you would need to use the `cuda_add_executable` and `cuda_add_library` macros -- you can't use CUDA with standard CMake add_library().  However, your buildscript does add .cu files directly to source lists and build them with add_library().  So, you're actually using the CMake CUDA language, not FindCUDA.

For one thing, this means that your actual minimum CMake version is 3.10, not 3.8.1 like it says in your cmake_minimum_required() call.   Also, it means that there is no need to call `find_package(CUDA)` as long as you add it to the list of languages.

Reply:
Right below this I have `enable_language(CUDA)`. Shall I also remove it, since I'm going to add `cuda` to `enable_language()` in the main `CMakeLists.txt` per your first comment? Or which way is better(Originally I thought you only need to enable CUDA when you want to build the CUDA version)?

Reply:
If I commented out `find_package(CUDA)` and wherever I enabled CUDA, it always gives me errors like:

_CMakeFiles/quicklib.dir/octree/grid_packer.cpp.o: In function `gpu_get_pfbased_basis_function_lists_new_imp(std::vector<node, std::allocator<node> >*, std::vector<node, std::allocator<node> >*, std::vector<bflist, std::allocator<bflist> >*)':
grid_packer.cpp:(.text+0x5183): undefined reference to `gpu_get_octree_info'
CMakeFiles/quicklib.dir/modules/quick_api_module.f90.o: In function `__quick_api_module_MOD_delete_quick_job':
quick_api_module.f90:(.text+0x210): undefined reference to `gpu_shutdown_'_
...

Reply:
> Right below this I have enable_language(CUDA). Shall I also remove it, since I'm going to add cuda to enable_language() in the main CMakeLists.txt per your first comment?

Oh right I missed that.  Yeah it's easiest to just add it to the enable_languages() call.

Reply:
> If I commented out find_package(CUDA) and wherever I enabled CUDA, it always gives me errors like:

I wonder if there's some conditions that are getting turned off when FindCUDA isn't called...